src/shogun/clustering/KMeansBase.cpp

```diff
@@ -163,18 +162,18 @@ void KMeansBase::initialize_training(const std::shared_ptr<Features>& data)
 
 	/* if kmeans++ to be used */
 	if (use_kmeanspp)
-		mus_initial=kmeanspp();
+		initial_centers = kmeanspp();
 
 	R=SGVector<float64_t>(k);
 
-	mus=SGMatrix<float64_t>(dimensions, k);
+	cluster_centers = SGMatrix<float64_t>(dimensions, k);
 	/* cluster_centers=zeros(dimensions, k) ; */
-	memset(mus.matrix, 0, sizeof(float64_t)*centers_size);
+	memset(cluster_centers.matrix, 0, sizeof(float64_t) * centers_size);
 
-	if (mus_initial.matrix)
+	if (initial_centers.matrix)
```

Comment:
> We need to find a way to initialise KMeans with clusters centres provided by the user using something like .put("initial_clusters"). Then when KMeans runs it will either initialise cluster centres itself or use the user provided values.

@gf712 i guess this is what you want?

Reply:
yup, exactly, thanks!